pkg/auth/credentials.go

```diff
@@ -109,11 +125,35 @@ func GetNewCredentials() (cred Credentials, err error) {
 	if err != nil {
 		return cred, err
 	}
-	cred.SecretKey = string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen])
+	cred.SecretKey = string([]byte(base64.URLEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen])
+
+	if _, ok := m["exp"]; !ok {
+		cred.Expiration = timeSentinel
+		return cred, nil
+	}
```

Comment:
The above 4 lines are redundant because of the code that follows

Reply:
why? if there is no `exp` then we don't need to set the expiration. We don't want sessionToken for credentials without claims .. etc.